ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/trielog/TrieLogLayer.java

```diff
@@ -98,10 +98,7 @@ public TrieLogLayer addAccountChange(
   public TrieLogLayer addCodeChange(
       final Address address, final Bytes oldValue, final Bytes newValue, final Hash blockHash) {
     checkState(!frozen, "Layer is Frozen");
-    code.put(
-        address,
-        new BonsaiValue<>(
-            oldValue == null ? Bytes.EMPTY : oldValue, newValue == null ? Bytes.EMPTY : newValue));
+    code.put(address, new BonsaiValue<>(oldValue, newValue, newValue == null));
```

Comment:
@matkt I believe we were previously returning Bytes.EMPTY here because the old BonsaiLayeredWorldState would serve state directly from TrieLogs.  

Since the 5123 bonsai refactor, we should be able to treat this consistently as null.  Otherwise a constructed trielog != that same trielog serialized and then deserialized.  It was the difference between code == null vs code == 0x.